login/service.go

```diff
@@ -331,7 +321,12 @@ func (keycloak *KeycloakOAuthProvider) AuthCodeCallback(ctx *app.CallbackAuthori
 		return nil, err
 	}
 
-	redirectTo := referrerURL.String() + "?code=" + ctx.Code + "&state=" + ctx.State.String()
+	parameters := url.Values{}
```

Comment:
`parameters := referrerURL.Query()`
Otherwise you will lost existing params in referrerURL.